src/plugins/reticle/reticle.game.ts

```diff
@@ -0,0 +1,61 @@
+import RAPIER from "@dimforge/rapier3d-compat";
+import { defineComponent, defineQuery, enterQuery, exitQuery, addComponent, removeComponent } from "bitecs";
+import { vec3, quat, mat4 } from "gl-matrix";
+import { Vector3 } from "three";
+
+import { Transform } from "../../engine/component/transform";
+import { GameState } from "../../engine/GameTypes";
+import { getModule, Thread } from "../../engine/module/module.common";
+import { PhysicsModule } from "../../engine/physics/physics.game";
+import { ReticleFocusMessage } from "./reticle.common";
+
+const FocusComponent = defineComponent();
+const focusQuery = defineQuery([FocusComponent]);
+const enterFocusQuery = enterQuery(focusQuery);
+const exitFocusQuery = exitQuery(focusQuery);
+
+const MAX_FOCUS_DISTANCE = 10;
+
+const _target = vec3.create();
+const _cameraWorldQuat = quat.create();
+
+const collisionGroups = 0x00f0_000f;
+
+export function ReticleFocusSystem(ctx: GameState) {
+  const physics = getModule(ctx, PhysicsModule);
+
+  // raycast outward from camera
+  const cameraMatrix = Transform.worldMatrix[ctx.activeCamera];
+  mat4.getRotation(_cameraWorldQuat, cameraMatrix);
+
+  const target = vec3.set(_target, 0, 0, -1);
+  vec3.transformQuat(target, target, _cameraWorldQuat);
+  vec3.scale(target, target, MAX_FOCUS_DISTANCE);
+
+  const source = mat4.getTranslation(vec3.create(), cameraMatrix);
+
+  const s: Vector3 = new Vector3().fromArray(source);
+  const t: Vector3 = new Vector3().fromArray(target);
+
+  const ray = new RAPIER.Ray(s, t);
+  const solid = true;
+  const maxToi = 4.0;
```

Comment:
The maxToi or the final ray length should be used to determine if you can actually interact with an object based on its max interaction distance.